src/js/components/PageHeaderBreadcrumbs.js

```diff
@@ -1,30 +1,58 @@
+import {Link} from 'react-router';
 import React from 'react';
+import {Tooltip} from 'reactjs-components';
 
 import Icon from './Icon';
 
 class PageHeaderBreadcrumbs extends React.Component {
-
-  getCaret(index) {
+  getCaret(key) {
     return (
       <li className="page-header-breadcrumb page-header-breadcrumb-caret"
-        key={`caret-${index}`}>
+        key={`caret-${key}`}>
         <Icon color="light-grey" id="caret-right" size="mini" />
       </li>
     );
   }
 
   render() {
-    const {props: {iconID, breadcrumbs}} = this;
-
+    const {props: {breadcrumbs, iconID, iconRoute}} = this;
+    const breadcrumbCount = breadcrumbs.length;
     const containerIcon = (
       <li
         className="page-header-breadcrumb page-header-breadcrumb-icon"
         key="-1">
-        <Icon family="product" id={iconID} size="small" />
+        <Link to={iconRoute}>
+          <Icon family="product" id={iconID} size="small" />
+        </Link>
       </li>
     );
+    const shouldTruncateBreadcrumbs = breadcrumbCount > 3;
 
     const breadcrumbElements = breadcrumbs.reduce((memo, breadcrumb, index) => {
+      if (index === 0) {
+        memo.push(this.getCaret('first-caret'));
+      }
+
+      if (shouldTruncateBreadcrumbs && index === breadcrumbCount - 3) {
+        memo.push(
+          this.getCaret('first-caret-forreal'),
```

Comment:
Ughhh thank you for catching this, I did not intend to actually commit this, but here we are... 😳